WcaOnRails/app/models/registration.rb

```diff
@@ -24,12 +27,30 @@ class Registration < ActiveRecord::Base
     end
   end
 
-  def pending?
-    accepted_at.nil?
+  def deleted?
+    !deleted_at.nil?
   end
 
   def accepted?
-    !pending?
+    !deleted? && !accepted_at.nil?
```

Comment:
Well, it might be okay for this method to return `nil`, since nil is "falsey" in ruby. However, there may be valid reasons for wanting it to be a boolean. One way to coerce `accepted_at` to a boolean is to do `!!accepted_at`.